File: app/src/main/java/com/futsch1/medtimer/reminders/scheduling/CyclesHelper.java
package com.futsch1.medtimer.reminders.scheduling;

import android.annotation.SuppressLint;

import com.futsch1.medtimer.database.Reminder;

import java.time.LocalDate;

public class CyclesHelper {

    private CyclesHelper() {
        // Intentionally empty
    }

    @SuppressLint("DefaultLocale")
    public static String getCycleCountString(Reminder reminder) {
        if (reminder.getPauseDays() != 0 && reminder.getConsecutiveDays() != 1) {
            long cycleStartDay = reminder.getCycleStartDay();
            long dayInCycle = LocalDate.now().toEpochDay() - cycleStartDay;
            int cycleLength = reminder.getConsecutiveDays() + reminder.getPauseDays();

            return String.format(" (%d/%d)", dayInCycle % cycleLength + 1, reminder.getConsecutiveDays());
        } else {
            return "";
        }
    }
}

File: app/src/main/java/com/futsch1/medtimer/reminders/scheduling/CyclesHelper.kt
package com.futsch1.medtimer.reminders.scheduling

import android.annotation.SuppressLint
import com.futsch1.medtimer.database.Reminder
import java.time.LocalDate

object CyclesHelper {
    @SuppressLint("DefaultLocale")
    fun getCycleCountString(reminder: Reminder): String {
        if (reminder.pauseDays == 0 || reminder.consecutiveDays == 1) {
            return ""
        }

        val cycleStartDay = reminder.cycleStartDay
        val dayInCycle = LocalDate.now().toEpochDay() - cycleStartDay
        val cycleLength = reminder.consecutiveDays + reminder.pauseDays

        val dayWithinCycle = dayInCycle % cycleLength + 1
        return " ($dayWithinCycle/${reminder.consecutiveDays})"
    }
}
